helm-chart/crds/meshconfig.yaml

```diff
@@ -0,0 +1,182 @@
+                    pattern: ^[1-9][0-9]{0,5}(h|m)$
+                    type: string
+                type: object
+              nginxErrorLogLevel:
+                description: NGINXErrorLogLevel is the NGINX error log level.
+                enum:
+                - debug
+                - info
+                - notice
+                - warn
+                - error
+                - crit
+                - alert
+                - emerg
+                type: string
+              nginxLBMethod:
+                description: NGINXLBMethod is the NGINX load balancing method.
+                enum:
+                - least_conn
+                - least_time
+                - least_time last_byte
+                - least_time last_byte inflight
+                - random
+                - random two
+                - random two least_conn
+                - random two least_time
+                - random two least_time=last_byte
+                - round_robin
+                type: string
+              nginxLogFormat:
+                description: NGINXLogFormat is the NGINX log format.
+                enum:
+                - default
+                - json
+                type: string
+              prometheusAddress:
+                description: PrometheusAddress is the address of a Prometheus server
+                  deployed in your Kubernetes cluster.
+                type: string
+              telemetry:
+                description: Telemetry is the configuration for telemetry.
+                properties:
+                  exporters:
+                    description: Exporters is the exporters configuration for telemetry.
+                    properties:
+                      otlp:
+                        description: Otlp is the configuration for an OTLP gRPC exporter.
+                        properties:
+                          host:
+                            description: Host of the OpenTelemetry gRPC exporter to connect to.
+                            minLength: 1
+                            type: string
+                          port:
+                            description: Port of the OpenTelemetry gRPC exporter to connect to.
+                            format: int32
+                            maximum: 65535
+                            minimum: 0
+                            type: integer
+                        type: object
+                    required:
+                    - otlp
+                    type: object
+                  samplerRatio:
+                    description: SamplerRatio is the percentage of traces that are
+                      processed and exported to the telemetry backend.
+                    maximum: 1
+                    minimum: 0
+                    type: number
+                    format: float
+                type: object
+              transparent:
+                description: Transparent is set when the mesh is removed to turn all
```

Comment:
I see this field explicitly states not to edit. Is there a way to make this field immutable?

Reply:
It needs to be mutable because our CLI will be setting it when the mesh is removed.

Reply:
Okay, that makes sense. Would it be worth adding why this field should not be changed?

Reply:
I can do that.

Reply:
Actually I feel like the first sentence should be informative enough. Unless you had something else in mind?

Reply:
I just don't like having a lever to pull, but saying **never** to pull it. But I agree it may be outside of our supported patterns to change this value.

I guess it's more a question of if we want to present values in our API that customers should never change. Otherwise, we would want to say something more along the lines of, "Exercise caution: Make sure you fully understand the potential risks before editing this value."

Reply:
I agree it's not ideal to have this value, but we have to have a way to set this when we remove the mesh. I can add a statement like "For mesh controller use only".

Reply:
Actually....maybe we could set this in the ConfigMap, and then the controller reports the status back in the CRD so we know it's completed...

Reply:
For sure. I definitely like that wording better.

Reply:
If that works as expected it's a much better solution. I like the idea of keeping the controller only fields out of the customer api.